File: reactivestreams/src/main/kotlin/reactivestreams/r2dbc/service/AuthService.java
package reactivestreams.r2dbc.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import reactivestreams.r2dbc.domain.AuthRepository;
import reactor.core.publisher.Mono;

@RequiredArgsConstructor
@Service
public class AuthService {

    private final AuthRepository authRepository;

    public Mono<String> findnameByToken(String token) {
        return authRepository.findnameByToken(token).last();
    }

}

File: reactivestreams/src/main/kotlin/reactivestreams/coroutine/service/AuthService.kt
package reactivestreams.coroutine.service

import org.springframework.stereotype.Service
import reactivestreams.coroutine.domain.AuthRepository
import reactor.core.publisher.Mono

@Service
class AuthService(
    private val authRepository: AuthRepository,
) {

    fun findnameByToken(token: String): Mono<String> {
        return authRepository.findnameByToken(token).last()
    }

}